In [4]:
import json

In [5]:
import numpy as np

In [6]:
from pytube import YouTube

In [10]:
dataset=json.load(open("C:/Users/asd/Desktop/Research Work/videoAds/video/cleaned_result/video_Effective_clean.json"))

In [11]:
print(dataset['KONL05sae4E'])

2


In [12]:
print(dataset.values())

dict_values(['2', '5', '5', '2', '5', '2', '2', '5', '2', '3', '5', '5', '5', '5', '1', '1', '3', '3', '4', '5', '2', '3', '5', '5', '4', '1', '5', '5', '5', '2', '3', '3', '5', '5', '4', '3', '1', '5', '4', '5', '5', '5', '1', '5', '3', '3', '2', '3', '5', '5', '5', '3', '3', '4', '5', '5', '4', '3', '4', '3', '5', '5', '4', '3', '3', '5', '3', '5', '4', '3', '5', '5', '5', '4', '5', '3', '5', '3', '3', '4', '3', '5', '3', '5', '5', '3', '5', '5', '3', '3', '3', '3', '3', '1', '4', '3', '1', '3', '3', '4', '5', '2', '3', '3', '3', '3', '5', '2', '3', '3', '4', '5', '5', '4', '3', '3', '5', '3', '3', '3', '2', '3', '5', '5', '5', '3', '1', '5', '4', '5', '4', '3', '5', '3', '5', '3', '5', '5', '3', '3', '3', '3', '4', '3', '3', '3', '4', '3', '3', '5', '5', '4', '5', '4', '5', '5', '3', '3', '3', '4', '4', '5', '5', '3', '3', '2', '3', '5', '5', '3', '4', '5', '3', '5', '5', '4', '3', '3', '3', '5', '5', '3', '2', '5', '1', '2', '5', '3', '5', '3', '2', '3', '5', '4', '3', '4', '5', '2

In [21]:
dataset.keys()

dict_keys(['KONL05sae4E', 'VZ9lXgcYL50', 'Pzl86IjTpHI', 'XutXCTwwmYc', 'h6CcxJQq1x8', 'w22oOw3RIAk', 'XkgGZobFUq0', '1KkVdbUr250', 'D-Q5mUes_8k', 'bqmzpwxDJ9s', 'mcP_f1UEQVo', '-nYhixp7bXI', 'xtJqw--DGl8', 'qhbliUq0_r4', 'EIUSkKTUftU', 'QEDRDrV55q8', '5AP5zQrJRIg', 'DlB4hFTAxOo', 'PsH4uuRT2w8', 'sPneQzqhh4g', 'G2wBnjtmmg8', 'TL4LEfRr1TI', 'QNh18bRw_0k', 'MX145Tu4MHY', 'i6O98o2FRHw', '9IZ4CN-CTCM', '5AuLkMBAFZg', '-rMrkyMn-sY', 'iZIHSn1ILxs', 'w4kHmbZTvyI', 'e8iwjnN8j3A', '-S0Azz85mlA', 'gm_n76Dsl0c', 'BhHONpmlxPc', 'g7xSwrdkOMM', 'dNjIpip6e3A', 'B-XHdkCsc5U', '-wXjs6DXoeY', 'QXAYIYdvnys', 'hFYCjZvv8y4', '7GKuZu9xkj0', 'r3idOGxm3FA', 'n1ClxdeiIUg', 'lfABnAS5i9Q', 'X6pyDwa3nGw', 'rBey5TrvoDw', 'mulAi7cno2Y', '1jP4O7rEHQ8', '-iKqr230U9M', 'wTfBWIJ8xQo', '0tteHhYh9rU', 'z01w7kSkDbg', 'baicgyq5UQo', '3zQ1lFaQBVc', 'P9LHzVEPodg', '7jXUWe3MV_c', '-u8ejR98KRE', 'xeo9mVnuuWo', 'LL0s6qOIGr8', '-XidtaAetyQ', 'hy2rGDWgOu8', '8zOgbkg6Akw', '_wZSB0b9OCA', 'AEMK1O0BDqA', 'biEjueDCnqM', 'mh6vHoyBs58',

In [22]:
dataset['KONL05sae4E']

'2'

In [23]:
x=[]
y=[]
for item in dataset.items():
    x.append(item[0])
    y.append(item[1])
print(len(x))
#print(x)
print(len(y))
#print(y)

3477
3477


In [24]:
from collections import Counter
Counter(y)

Counter({'1': 193, '2': 261, '3': 1319, '4': 426, '5': 1278})

We have 3477 video links. I checked a few randomly. The first link's video is unavailable. Their is a chance that more such videos may exist in the databse. I need to find a way to check that. If we look at it as just a classification problem, we have 5 scores. The video dataset is not much equally distributed in that context. Also, the observed videos(randomly) are of various lengths, not any fixed duration. 

In [25]:
# import urllib.request
# urllib.request.urlretrieve('https://r2---sn-aigs6n7y.googlevideo.com/videoplayback?ratebypass=yes&signature=CC278A6170106BAEDA6EA4B09E6C2DD2AAFB0B23.D76B269503FD0749301E3CF1A9B78AEC33ECA264&gir=yes&mime=video%2Fmp4&key=yt6&requiressl=yes&clen=3440207&mm=31&mn=sn-aigs6n7y&id=o-AFwA5-69hHKsOO-R-znOd5OgrKA9UrTRWVkteBBGXv82&mt=1516429669&mv=u&sparams=clen%2Cdur%2Cei%2Cgir%2Cid%2Cip%2Cipbits%2Citag%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Cratebypass%2Crequiressl%2Csource%2Cexpire&ms=au&ip=149.13.117.213&ei=UONiWtuLJ56V1gLA6IJI&itag=18&pl=24&source=youtube&expire=1516451760&ipbits=0&dur=83.498&lmt=1393621001853388&video_id=sPneQzqhh4g&title=Shadow+Puppets', 'checkVideo.mp4')

A video can not be directly downloaded from youtube link used to access. The video here has been downloaded using ssyoutube. Not yet according to my work

In [27]:
yt = YouTube('https://www.youtube.com/watch?v='+x[2])
print(yt.title)
yt.streams.all()

RegexMatchError: regex pattern (\W[\'"]?t[\'"]?: ?[\'"](.+?)[\'"]) had zero matches

In [13]:
stream=yt.streams.filter(file_extension='mp4').all()
stream[0].download()

In [92]:
#print(x[0])
startInd=1000
endInd=2000
x_vid=[]
y_lab=[]
for i in range(startInd,endInd):
    check=0
    link= "https://www.youtube.com/watch?v="+x[i]
    try:
        yt= YouTube(link)
        check=1
    except:
        check=0
    if(check == 1):
        x_vid.append(yt.title)
        y_lab.append(y[i])
        #stream=yt.streams.filter(file_extension='mp4').all()
        #stream[0].download()
print(len(x_vid))
print(x_vid)
print(len(y_lab))
print(y_lab)

798
['Care Advantage Roanoke 15 Second 2015 Commercial', 'NFL on Xbox: Detention (& Madden) with Marshawn Lynch', 'Guinness Horse Surfer Ad', 'Center Fresh Chewing Gum', 'Mcdonalds Ronald Gram Commercial', 'Sparsh Baby Bar Ad - India', 'Origin Access feat. Bill Nye – Multiverse Theory', 'Summoners War - RIFT OF WORLDS', 'U.S. Bank: Start Smart Savings Program :15', 'Excedrin Extra Strength Geltabs Commercial: “Mom Has a Headache!”', 'Funny Golf commercial', 'Bud Light  |  "Clothing Drive" commercial', "Five Nights at Freddy's 4 Trailer", 'Baileys Cream With Spirit (:15 Commercial)', 'Israel Beyond Belief', 'Nothing Without It', 'Publix Pharmacy Medicare Part D - Make the Switch today!', '3 Day Blinds Custom Window Treatments', 'Winter Olympics 2014: Trailer - BBC Sport', 'Turkish Airlines - Drogba vs. Messi: #EpicFood (Official full length version)', 'NEW Visionnaire Nuit Advanced Multi-Correcting Gel-In-Oil', 'Look North (30-second version)', 'ON Amino Energy with Shaun Stafford', 'NE

In [93]:
x_vidReplace=x_vid
y_labReplace=y_lab

In [81]:
x_vid[799]
y_lab[799]

'3'

In [94]:
[x.encode('utf-8') for x in x_vid]

[b'Care Advantage Roanoke 15 Second 2015 Commercial',
 b'NFL on Xbox: Detention (& Madden) with Marshawn Lynch',
 b'Guinness Horse Surfer Ad',
 b'Center Fresh Chewing Gum',
 b'Mcdonalds Ronald Gram Commercial',
 b'Sparsh Baby Bar Ad - India',
 b'Origin Access feat. Bill Nye \xe2\x80\x93 Multiverse Theory',
 b'Summoners War - RIFT OF WORLDS',
 b'U.S. Bank: Start Smart Savings Program :15',
 b'Excedrin Extra Strength Geltabs Commercial: \xe2\x80\x9cMom Has a Headache!\xe2\x80\x9d',
 b'Funny Golf commercial',
 b'Bud Light  |  "Clothing Drive" commercial',
 b"Five Nights at Freddy's 4 Trailer",
 b'Baileys Cream With Spirit (:15 Commercial)',
 b'Israel Beyond Belief',
 b'Nothing Without It',
 b'Publix Pharmacy Medicare Part D - Make the Switch today!',
 b'3 Day Blinds Custom Window Treatments',
 b'Winter Olympics 2014: Trailer - BBC Sport',
 b'Turkish Airlines - Drogba vs. Messi: #EpicFood (Official full length version)',
 b'NEW Visionnaire Nuit Advanced Multi-Correcting Gel-In-Oil',
 b'Loo

In [95]:
fileTry = open('titles2000.txt', 'w',encoding='utf-8')
for item in x_vid:
    fileTry.write("%s\n" % item)
fileTry.close()

In [96]:
fileTry = open('labels2000.txt', 'w')
for item in y_lab:
    fileTry.write("%s\n" % item)
fileTry.close()

In [115]:
fileTry = open('titless.txt', 'r',encoding='utf-8')
lisst=fileTry.read()
downTitles=lisst.split('\n')

2730

In [116]:
fileTry = open('labelss.txt', 'r',encoding='utf-8')
lisst=fileTry.read()
downLabels=lisst.split('\n')

In [134]:
[a.encode('utf-8') for a in downLabels]

[b'\xef\xbb\xbf5',
 b'5',
 b'2',
 b'2',
 b'2',
 b'5',
 b'2',
 b'3',
 b'5',
 b'5',
 b'5',
 b'5',
 b'1',
 b'1',
 b'3',
 b'3',
 b'5',
 b'2',
 b'3',
 b'5',
 b'5',
 b'4',
 b'5',
 b'5',
 b'5',
 b'2',
 b'3',
 b'5',
 b'5',
 b'4',
 b'3',
 b'1',
 b'5',
 b'4',
 b'5',
 b'5',
 b'5',
 b'1',
 b'5',
 b'3',
 b'2',
 b'5',
 b'5',
 b'5',
 b'3',
 b'5',
 b'5',
 b'3',
 b'4',
 b'3',
 b'5',
 b'4',
 b'3',
 b'3',
 b'5',
 b'3',
 b'5',
 b'4',
 b'3',
 b'5',
 b'5',
 b'5',
 b'4',
 b'5',
 b'5',
 b'3',
 b'4',
 b'3',
 b'5',
 b'3',
 b'5',
 b'5',
 b'3',
 b'5',
 b'3',
 b'3',
 b'3',
 b'1',
 b'4',
 b'3',
 b'1',
 b'3',
 b'3',
 b'4',
 b'2',
 b'3',
 b'3',
 b'3',
 b'5',
 b'2',
 b'3',
 b'4',
 b'5',
 b'5',
 b'3',
 b'5',
 b'3',
 b'2',
 b'3',
 b'5',
 b'5',
 b'3',
 b'1',
 b'5',
 b'4',
 b'5',
 b'4',
 b'3',
 b'5',
 b'3',
 b'3',
 b'5',
 b'3',
 b'3',
 b'3',
 b'3',
 b'4',
 b'3',
 b'3',
 b'4',
 b'3',
 b'3',
 b'5',
 b'5',
 b'4',
 b'5',
 b'4',
 b'5',
 b'3',
 b'3',
 b'4',
 b'4',
 b'5',
 b'5',
 b'3',
 b'2',
 b'5',
 b'5',
 b'3',
 b'4',
 b'3',
 

In [139]:
downLabels[0]

'\ufeff5'

In [120]:
count=[0,0,0,0,0]

In [140]:
for i in range (np.size(downLabels)):
    if downLabels[i]=='1':
        count[0]=count[0]+1
    elif downLabels[i]=='2':
        count[1]=count[1]+1
    elif downLabels[i]=='3':
        count[2]=count[2]+1
    elif downLabels[i]=='4':
        count[3]=count[3]+1
    elif downLabels[i]=='5':
        count[4]=count[4]+1

In [141]:
print(count)

[159, 213, 968, 339, 1050]


In [22]:
import numpy as np
print(np.size(y_lab))

194


In [13]:
from __future__ import unicode_literals
import youtube_dl
import urllib
import shutil
ydl_opts = {}
with youtube_dl.YoutubeDL(ydl_opts) as ydl:
    ydl.download(['https://www.youtube.com/watch?v=sPneQzqhh4g'])

[youtube] sPneQzqhh4g: Downloading webpage
[youtube] sPneQzqhh4g: Downloading video info webpage
[youtube] sPneQzqhh4g: Extracting video information
[download] Shadow Puppets-sPneQzqhh4g.webm has already been downloaded
[download] 100% of 1.97MiB


In [35]:
try:
    yt2 = YouTube('https://www.youtube.com/watch?v=sPneQzqhh4g')
except:
    print("Nahin exist karti!")
print(yt2.title)
yt2.streams.all()

Shadow Puppets


[<Stream: itag="43" mime_type="video/webm" res="360p" fps="30fps" vcodec="vp8.0" acodec="vorbis">,
 <Stream: itag="18" mime_type="video/mp4" res="360p" fps="30fps" vcodec="avc1.42001E" acodec="mp4a.40.2">,
 <Stream: itag="36" mime_type="video/3gpp" res="240p" fps="30fps" vcodec="mp4v.20.3" acodec="mp4a.40.2">,
 <Stream: itag="17" mime_type="video/3gpp" res="144p" fps="30fps" vcodec="mp4v.20.3" acodec="mp4a.40.2">,
 <Stream: itag="133" mime_type="video/mp4" res="240p" fps="30fps" vcodec="avc1.4d400d">,
 <Stream: itag="160" mime_type="video/mp4" res="144p" fps="30fps" vcodec="avc1.4d400c">,
 <Stream: itag="140" mime_type="audio/mp4" abr="128kbps" acodec="mp4a.40.2">,
 <Stream: itag="171" mime_type="audio/webm" abr="128kbps" acodec="vorbis">,
 <Stream: itag="249" mime_type="audio/webm" abr="50kbps" acodec="opus">,
 <Stream: itag="250" mime_type="audio/webm" abr="70kbps" acodec="opus">,
 <Stream: itag="251" mime_type="audio/webm" abr="160kbps" acodec="opus">]